### ISSUE TITLE
Rule proposal: `no-array-for-each`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#955]
`for...of` is more readable.

## Fail

```js
foo.forEach(x => callback(x));
```

## Pass

```js
for (const x of foo) {
	callback(x);
}
```

**Comment by silverwind** [comment]
Maybe also cover `for...in` and name it `prefer-for-of`?

**Comment by sindresorhus** [comment]
`for...in` is covered by another rule here.

**Comment by fisker** [comment]
The auto-fix should replace `return` to `break`, but if `return` is in a loop,

Example:

```js
foo.forEach(function (x) {
	for (let i = 0; i < foo.length; i ++) {
		if (i === 2) {
			return;
		}
	}
});
```

```js
for (const x of foo) {
	for (let i = 0; i < foo.length; i ++) {
		if (i === 2) {
			break; // This won't break outer loop
		}
	}
}
```

Should we just ignore this case?

**Comment by sindresorhus** [comment]
> Should we just ignore this case?

Yes. It should be reported, but not auto-fixed.

**Comment by fisker** [comment]
GitHub disallow this too https://github.com/github/eslint-plugin-github/blob/main/docs/rules/array-foreach.md

**Comment by sindresorhus** [comment]
This rule is now accepted.

**Comment by fregante** [comment]
For reference, there's a `prefer-for-of` too, but it's against regular `for` loops: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-for-of.md
